src/ai/backend/manager/api/admin.py

```diff
@@ -314,6 +322,123 @@ async def shutdown(app: web.Application) -> None:
     pass
 
 
+@auth_required
+async def handle_gql_ws(request: web.Request) -> web.WebSocketResponse:
+    ws = web.WebSocketResponse(protocols=["graphql-transport-ws", "graphql-ws"])
+    await ws.prepare(request)
+
+    # Create context once
+    root_ctx: RootContext = request.app["_root.context"]
+    processors_ctx = await ProcessorsCtx.from_request(request)
+    context = StrawberryGQLContext(
+        processors=processors_ctx.processors,
+        config_provider=root_ctx.config_provider,
+        event_hub=processors_ctx.event_hub,
+        event_fetcher=processors_ctx.event_fetcher,
+        valkey_bgtask=processors_ctx.valkey_bgtask,
+    )
+
+    schema = request.app["admin.context"].gql_v2_schema
+
+    async for msg in ws:
+        if msg.type == web.WSMsgType.TEXT:
+            data = msg.json()
+
+            if data.get("type") == "connection_init":
+                await ws.send_str('{"type":"connection_ack"}')
+
+            elif data.get("type") == "subscribe":
+                subscription_id = data.get("id")
+                payload = data.get("payload", {})
```

Comment:
Please check the format of how values are passed from GraphQL and define it with Pydantic.